stl/inc/deque

```diff
@@ -1282,6 +1379,71 @@ public:
         return begin() + static_cast<difference_type>(_Off);
     }
 
+#ifdef __cpp_lib_containers_ranges
+    template <_Container_compatible_range<_Ty> _Rng>
+    iterator insert_range(const_iterator _Where, _Rng&& _Range) {
+        size_type _Off = static_cast<size_type>(_Where - begin());
+
+#if _ITERATOR_DEBUG_LEVEL == 2
+        _STL_VERIFY(_Mysize() >= _Off, "deque insert iterator outside range");
+#endif // _ITERATOR_DEBUG_LEVEL == 2
+
+        size_type _Oldsize = _Mysize();
+
+        auto _UFirst = _RANGES _Ubegin(_Range);
+        auto _ULast  = [&] {
+            if constexpr (_RANGES bidirectional_range<_Rng>) {
+                return _RANGES _Get_final_iterator_unwrapped(_Range);
+            } else {
+                return _RANGES _Uend(_Range);
+            }
+        }();
+        if (_UFirst != _ULast) {
+            _Orphan_all();
+            if (_Off <= _Mysize() / 2) { // closer to front, push to front then rotate
+                _TRY_BEGIN
+                if constexpr (_RANGES bidirectional_range<_Rng>) {
+                    while (_UFirst != _ULast) {
+                        _Emplace_front_internal(*--_ULast);
+                    }
+                } else {
+                    for (; _UFirst != _ULast; ++_UFirst) {
+                        _Emplace_front_internal(*_UFirst); // prepend flipped
+                    }
+                }
+                _CATCH_ALL
+                while (_Oldsize < _Mysize()) {
+                    pop_front(); // restore old size, at least
+                }
+                _RERAISE;
+                _CATCH_END

```

Comment:
```suggestion
                _Restore_old_size_guard<_Pop_direction::_Front> _Guard{this, _Oldsize};
                if constexpr (_RANGES bidirectional_range<_Rng>) {
                    while (_UFirst != _ULast) {
                        _Emplace_front_internal(*--_ULast);
                    }
                } else {
                    for (; _UFirst != _ULast; ++_UFirst) {
                        _Emplace_front_internal(*_UFirst); // prepend flipped
                    }
                }
```

Reply:
Thanks! I completely missed this out. Taking both of these, and making similar fixes to `append_range` and `prepend_range`. While I'm at it, factoring out the commonality in `insert(iter, first, last)` and `insert_range(iter, range)` which I failed to do earlier.